temporalio/client.py

```diff
@@ -1984,114 +2095,1656 @@ async def __anext__(self) -> temporalio.api.history.v1.HistoryEvent:
+        ...
+
+    @overload
+    async def memo_value(
+        self, key: str, default: AnyType, *, type_hint: Type[ParamType]
+    ) -> Union[AnyType, ParamType]:
+        ...
+
+    async def memo_value(
+        self,
+        key: str,
+        default: Any = temporalio.common._arg_unset,
+        *,
+        type_hint: Optional[Type] = None,
+    ) -> Any:
+        """Memo value for the given key, optional default, and optional type
+        hint.
+
+        Args:
+            key: Key to get memo value for.
+            default: Default to use if key is not present. If unset, a
+                :py:class:`KeyError` is raised when the key does not exist.
+            type_hint: Type hint to use when converting.
+
+        Returns:
+            Memo value, converted with the type hint if present.
+
+        Raises:
+            KeyError: Key not present and default not set.
+        """
+        payload = self.raw_entry.memo.fields.get(key)
+        if not payload:
+            if default is temporalio.common._arg_unset:
+                raise KeyError(f"Memo does not have a value for key {key}")
+            return default
+        return (
+            await self.data_converter.decode(
+                [payload], [type_hint] if type_hint else None
+            )
+        )[0]
+
+
+@dataclass
+class ScheduleListSchedule:
+    """Details for a listed schedule.
+
+    .. warning::
+        Schedules are an experimental feature.
+    """
+
+    action: ScheduleListAction
+    """Action taken when scheduled."""
+
+    spec: ScheduleSpec
+    """When the action is taken."""
+
+    state: ScheduleListState
+    """State of the schedule."""
+
+    @staticmethod
+    def _from_proto(
+        info: temporalio.api.schedule.v1.ScheduleListInfo,
+    ) -> ScheduleListSchedule:
+        # Only start workflow supported for now
+        if not info.HasField("workflow_type"):
+            raise ValueError("Unknown action on schedule")
+        return ScheduleListSchedule(
+            action=ScheduleListActionStartWorkflow(workflow=info.workflow_type.name),
+            spec=ScheduleSpec._from_proto(info.spec),
+            state=ScheduleListState._from_proto(info),
+        )
+
+
+class ScheduleListAction(ABC):
+    """Base class for an action a listed schedule can take.
+
+    .. warning::
+        Schedules are an experimental feature.
+    """
+
+    pass
+
+
+@dataclass
+class ScheduleListActionStartWorkflow(ScheduleListAction):
+    """Action to start a workflow on a listed schedule.
+
+    .. warning::
+        Schedules are an experimental feature.
+    """
+
+    workflow: str
+    """Workflow type name."""
+
+
+@dataclass
+class ScheduleListInfo:
+    """Information about a listed schedule.
+
+    .. warning::
+        Schedules are an experimental feature.
+    """
+
+    recent_actions: Sequence[ScheduleActionResult]
+    """10 most recent actions, oldest first."""
```

Comment:
These both should be 5

Reply:
:+1: Then you need to update https://pkg.go.dev/go.temporal.io/sdk/internal#ScheduleInfo too

Reply:
https://github.com/temporalio/api/blob/44e4397262d480b9e6f17771be9ae1812d97384e/temporal/api/schedule/v1/message.proto#L332-L336 also says "ten" and https://typescript.temporal.io/api/namespaces/client#scheduledescription says "10". I will leave here and we can update all docs together.

Reply:
:+1: I will remove the counts on the list one. But we need to update https://typescript.temporal.io/api/interfaces/client.ScheduleSummary#info too.

Reply:
I removed the exact count on both update and describe in TS.

I know the spec mention 10 on ScheduleInfo, but given that this was made tweakable on the server, I think that we'd better consider that number as an implementation detail rather than a contract.